### PR TITLE
[ADD] foodhub_product_fields

### PR DESCRIPTION
**Pull request by victor-champonnois**
## Description
Ongoing discussion with client about who deals with the DB migration

## Odoo task (if applicable)
https://gestion.coopiteasy.be/web#id=9513&model=project.task&view_type=form&menu_id=

## Checklist before approval

- [ ] Tests are present (or not needed).
- [ ] Credits/copyright have been changed correctly.
- [ ] Change log snippet is present.
- [ ] (If a new module) Moving this to OCA has been considered.

### CODE REVIEW

**Review by carmenbianca — COMMENTED** [review]
This seems to have a lot of overlap with foodhub_copy_product_fields.

For consistency between the two modules, maybe it's a good idea to unite them under a single module? `foodhub_product_fields` or some such, which sets `copy` and `required` to True for some product fields.

**Comment by victor-champonnois** [comment]
@carmenbianca It could have been convenient if we had the idea before the development. Now that it's done I don't think it is worth it. What I remember we decided in CC is one custom module per demand.

**Comment by carmenbianca** [comment]
> What I remember we decided in CC is one custom module per demand.

Per functionality :) These functionalities seem _incredibly_ intertwined to me, because they touch (almost) the exact same fields.

The danger here is that if the client wants to set another field as `required`, they _probably_ implicitly also want `copy` to be set to True. This is easily overlooked when the two modules are separate. By merging the two modules, this discrepancy is easier to spot at a glance.

Merging the two modules should be fairly easy. Rename the module in this PR to `foodhub_product_fields`. Set `copy=True` for all the fields in this module that intersect with the fields in `foodhub_copy_product_fields`. Clear out `foodhub_copy_product_fields` and set it to depend on the new `foodhub_product_fields`. Make sure to document properly in `readme/DESCRIPTION.rst`. (As a matter of fact, I think `foodhub_copy_product_fields`'s documentation is already discrepant. Oops.)

I'll defer to another reviewer to determine whether this is the correct thing to do.

**Comment by victor-champonnois** [comment]
@carmenbianca OK I think you're right. I did the refactoring.

File: foodhub_product_fields/models/product_template.py [hunk 1]
```diff
@@ -1,16 +1,24 @@
...
-    weight = fields.Float(copy=True)
+    # All below fields have required=True added to them.
+    description_sale = fields.Text(required="True")
+    public_categ_ids = fields.Many2many(required="True")
+    pos_categ_id = fields.Many2one(required="True")
+    list_price = fields.Float(required="True")
+
+    # The below field have copy=True added to them.
+    origin_country_id = fields.Many2one(
+        "res.country",
```
Comment (carmenbianca) [inline foodhub_product_fields/models/product_template.py]: Apparently the module name here can be removed.

File: foodhub_product_fields/models/product_template.py [hunk 2]
```diff
@@ -19,14 +27,12 @@ class ProductTemplate(models.Model):
     seller_ids = fields.One2many(
         "product.supplierinfo",
         "product_tmpl_id",
```
Comment (carmenbianca) [inline foodhub_product_fields/models/product_template.py]: idem

File: foodhub_product_fields/models/product_template.py [hunk 3]
```diff
@@ -19,14 +27,12 @@ class ProductTemplate(models.Model):
...
+        required="True",
         copy=True,
     )
-    default_code = fields.Char(copy=True)
+    default_code = fields.Char(required="True", copy=True)
     hs_code_id = fields.Many2one(
         "hs.code",
```
Comment (carmenbianca) [inline foodhub_product_fields/models/product_template.py]: idem

**Review by carmenbianca — APPROVED** [review]
Awesome PR!

File: foodhub_product_fields/models/product_template.py (hunk 1 above)
Comment (victor-champonnois) [reply foodhub_product_fields/models/product_template.py]: Hmm, what module name ?? ^^

Comment (carmenbianca) [reply foodhub_product_fields/models/product_template.py]: Sorry, model name. res.country.

**Pull request merged**